.github/workflows/deploy-prerelease.yml

```
@@ -0,0 +1,224 @@
+    needs: check-dispatch-token
+    # Don't run if the event is triggered by a repository dispatch with a non-valid token, 
+    # but trigget if the event is not a repository dispatch
+    if: ${{ !cancelled() && !(github.event_name == 'repository_dispatch' && needs.check-dispatch-token.result != 'success') }}
+    outputs:
+      full-name: ${{ steps.get-env-name.outputs.full-name }}
+      dependecy-names: ${{ steps.get-dependencies.outputs.dependecy-names }}
+      pr-numbers: ${{ steps.get-dependencies.outputs.pr-numbers }}
+    env:
+      GH_TOKEN: ${{ github.token }}
+      DEPENDENCY_REPOS: replace_landsurface era5_grib_parallel
+      ARCH: noarch
+      CHANNEL_SUFFIX: _channel
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get environment name
+        id: get-env-name
+        run: echo "full-name=$(yq '.name' < ${{env.ENV_NAME}})-${{env.VERSION}}" >> $GITHUB_OUTPUT
+
+      # Search for the latest open PR (not a draft) in the replace_landsurface and era5_grib_parallel repos with
+      # changes in specific files. If a PR is found, download the artifact from the PR latest 
+      # (successful) workflow run. If a PR is not found, download the lates released version.
+      - name: Get latest dependencies
+        id: get-dependencies
+        run: |
+          download_latest_release() {
+            gh release download $(gh release view --repo ${{ github.repository_owner }}/$1 --json tagName --jq '.tagName') \
+            --repo ${{ github.repository_owner }}/$1 --pattern *.tar.bz2 -D ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}} && \
+            echo "${1}: Successfully downloaded latest release: \"$(basename ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/*.tar.bz2)\"."
+          }
+          download_run_artifact() {
+            (gh run download $run_id --repo ${{ github.repository_owner }}/$1 -D ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}} && \
+            echo "${1}: Successfully downloaded artifact \"$(basename ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/artifact/*.tar.bz2)\"." && \
+            mv ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}/artifact/*.tar.bz2 ${1}${{env.CHANNEL_SUFFIX}}/${{env.ARCH}}) || \
+            (echo "${1}: No valid artifact found. Dependency will be installed from the latest release.";\
+            download_latest_release ${1})
+          }
+
+
+          for dependency_repo in ${{env.DEPENDENCY_REPOS}}; do
+            echo "Getting latest \"${dependency_repo}\" dependency ..."
+    
+            # Get sha and number of latest open PR that changes either of the following files:
+            # [src/**, setup.py, pyproject.toml, .conda/**]
+            pr=$(gh pr list --repo ${{ github.repository_owner }}/${dependency_repo} --state open --draft=False \
+            --json headRefOid,files,number,url --jq '.[] | select(.files[].path | 
+            (startswith("src/") or (. == "setup.py") or (. == "pyproject.toml") or (startswith(".conda/"))))' \
+            | head -n 1)
```

If this could be formatted a bit more it might be a bit more readable, rather than having it all on the same line. Something like:
```suggestion
            pr=$(gh pr list --repo ${{ github.repository_owner }}/${dependency_repo} --state open --draft=False \
                --json headRefOid,files,number,url 
                --jq '.[] | select(.files[].path | 
                  (startswith("src/") or (. == "setup.py") or (. == "pyproject.toml") or (startswith(".conda/"))))' \
            | head -n 1)
```